### ISSUE TITLE
[Question]I would like an explanation of MMVMData.

### ISSUE THREAD

**Issue by 08D20088** [zhouyiks/CoLVA#4]
@zhouyiks 
I want to fine-tune the pure Qwen2.5VL model using MMVMData.

Qwestion1:
From reading the paper, I understand that MMVMData is a multiple-choice question dataset. However, when I look at the `mllm_match_choices_sft.json` and `mllm_match_reasoning_sft.json` files, they appear to be in a different format from the MMVM Benchmark.
So, I'm not sure how to convert this data to following format.
Please explain the format of the data.

```
{
    "images": ["cats/001.jpg", "cats/002.jpg"],
    "conversations": [
        {
            "from": "human",
            "value": "<image>\n<image>\nWhat are the differences between these two cats?"
        },
        {
            "from": "gpt",
            "value": "The first cat is an orange tabby with short fur and green eyes, while the second is a gray Siamese with blue eyes and pointed coloration. They also appear to be in different environments - the first is indoors on a couch, the second is outdoors in a garden."
        }
    ]
}
```

Qwestion2:
Also, what is `sa1b_coco_fmt_iminfo_500k_redirected.json`?
I would like to know.

**Comment by zhouyiks** [comment]
Answer to Question2: `sa1b_coco_fmt_iminfo_500k_redirected.json` lists 500k SA1B images which are used to train the RADIO Adapter in the pre-training phrase: https://github.com/zhouyiks/CoLVA/blob/e0d99c10b80a34b7763c5896b4a7aadd0dc7b600/projects/colva/configs/pretrain/qwen2vl_2b_radio_align_pretrain.py#L39

Answer to Question1: Yes, the latest version of our MMVMBench is constructed as a VQA dataset, which means we don't prompt the MLLMs with options of candidates' ID. Instead, we ask the MLLMs to recognize the integer ID of positive candidate and response with it directly, which is more reasonable compared to the older one (MCQ version). And the VQA version of MMVMBench is supported by official repository of open-compass/VLMEvalKit. 

For your requirements, I recommend using `mllm_match_reasoning_sft.json`. One sample look likes this: 
`{"id": "00000001", "file_names": ["./data/video_dataset/ytvis_2021/train/JPEGImages/dd1f8f53d6/00000.jpg", "./data/video_dataset/ytvis_2021/train/JPEGImages/dd1f8f53d6/00075.jpg"], "is_disappear": true, "pos_annotations": [[{"counts": [167886, 3, 355, 7, 353, 13, 347, 17, 343, 18, 342, 19, 341, 21, 339, 22, 338, 23, 339, 21, 341, 19, 343, 16, 350, 8, 576], "size": [360, 480]}], []], "neg_annotations": [[], [{"counts": [9860, 18, 341, 22, 335, 26, 26, 7, 300, 29, 22, 9, 297, 35, 18, 10, 30, 13, 253, 47, 1, 16, 28, 16, 250, 66, 15, 30, 247, 113, 246, 115, 244, 116, 243, 117, 243, 117, 242, 118, 241, 119, 241, 119, 241, 119, 241, 119, 241, 119, 240, 102, 4, 14, 225, 113, 11, 11, 223, 111, 15, 11, 221, 110, 18, 11, 221, 103, 26, 9, 222, 97, 32, 9, 222, 94, 36, 6, 223, 94, 37, 6, 223, 95, 265, 95, 28, 6, 231, 95, 26, 8, 231, 97, 21, 12, 230, 99, 18, 13, 230, 131, 229, 131, 229, 131, 230, 20, 1, 109, 231, 19, 3, 109, 230, 16, 5, 109, 231, 14, 7, 108, 232, 11, 10, 107, 254, 106, 256, 104, 258, 102, 259, 91, 1, 9, 292, 39, 21, 8, 293, 30, 29, 8, 297, 22, 34, 6, 298, 19, 37, 6, 300, 11, 349, 10, 145617], "size": [360, 480]}, {"counts": [92707, 6, 352, 9, 348, 15, 343, 17, 340, 20, 340, 14, 344, 15, 345, 12, 346, 13, 346, 13, 346, 14, 345, 13, 345, 15, 344, 14, 344, 15, 344, 15, 344, 15, 344, 15, 343, 16, 343, 16, 341, 19, 341, 17, 340, 20, 339, 19, 339, 21, 338, 20, 51, 11, 277, 19, 50, 15, 274, 21, 45, 22, 270, 20, 41, 29, 268, 22, 39, 33, 265, 21, 37, 37, 265, 27, 25, 43, 264, 29, 24, 45, 261, 38, 10, 52, 260, 100, 259, 101, 259, 102, 257, 103, 241, 12, 3, 105, 238, 123, 236, 124, 234, 126, 233, 127, 232, 128, 231, 110, 3, 16, 1, 10, 219, 111, 4, 15, 1, 11, 218, 111, 4, 27, 217, 112, 5, 26, 215, 113, 8, 24, 215, 113, 10, 22, 214, 114, 9, 23, 214, 114, 8, 24, 214, 114, 6, 26, 214, 114, 6, 25, 215, 114, 3, 28, 215, 143, 217, 142, 219, 136, 224, 135, 225, 135, 226, 135, 226, 134, 227, 23, 1, 112, 225, 22, 1, 113, 224, 20, 4, 112, 226, 17, 5, 115, 226, 11, 9, 115, 245, 116, 246, 114, 247, 129, 233, 127, 234, 126, 236, 124, 237, 123, 242, 62, 14, 43, 241, 49, 32, 38, 242, 46, 39, 33, 244, 41, 42, 33, 248, 35, 47, 30, 252, 29, 50, 29, 333, 27, 334, 26, 336, 23, 337, 18, 50135], "size": [360, 480]}, {"counts": [39037, 4, 355, 5, 353, 7, 351, 9, 350, 7, 353, 6, 352, 6, 345, 14, 342, 18, 337, 23, 331, 29, 328, 31, 328, 31, 7, 3, 319, 41, 319, 41, 319, 41, 319, 32, 339, 10, 127658], "size": [360, 480]}, {"counts": [64596, 7, 353, 7, 353, 8, 352, 8, 352, 8, 352, 8, 353, 7, 353, 6, 105677], "size": [360, 480]}]], "description": "The second image does not contain the same object as the marked instance in the first image because it depicts different people engaged in different activities. The marked individual in the first image is captured in a specific kicking motion, which is not present in the second image.", "vprompt_type": "mask"}`
The `description` can be the GPT response in your format.

It is possible to convert each item to your format. But there are four concerns:
1. Is the images object-centric in your cases? The images in MMVMData isn't. So you need use `pos_annotations`  to draw visual prompts on the images. Here is my codes: https://github.com/zhouyiks/CoLVA/blob/e0d99c10b80a34b7763c5896b4a7aadd0dc7b600/projects/colva/dataset/Qwen2VLDataset.py#L437
2. Is the category needed to format the user question? There is not a `category` label in my annotation. Maybe you can use the "The object marked by cyan-color contours" to specify the target.
3. There are disappear cases in my annotations, which means the object is not appear in the second image. You can use `is_disappear` to filter such cases out.
4. Some cases maybe too easy to train your model, we select some hard cases by using latest powerful open-source VLMs. You can  download them at `https://huggingface.co/datasets/zhouyik/MMVMData/blob/main/mllm_match_choices_hard.json`. Hard cases in `mllm_match_reasoning_sft.json` can be picked out by verifying whether their IDs is in `mllm_match_choices_hard.json`.

**Comment by 08D20088** [comment]
Thank you for explaining in detail.
I appreciate it very much.

https://github.com/QwenLM/Qwen2.5-VL/tree/main/qwen-vl-finetune
The format I presented is from the Qwen project.
I am currently using this code in an ongoing project, so I would like to train the data in this format.
Additionally, for specific reasons, I would like to train the data in a multiple-choice format.
However, according to your response, does this mean that the MMVM dataset, i.e., the SFT data, has a different format from the MMVMBench?

I understood "The description can be the GPT response in your format" to mean that the description serves as the answer to the question.
However, I couldn’t find the question itself in "mllm_match_reasoning_sft.json".

1. Is the image object-centric in your cases?
   Does this mean that I need to annotate the images before SFT using the code you provided?
   Also, based on `'./data/video_dataset/ytvis_2021/train/JPEGImages/dd1f8f53d6/00000.jpg'`, I assume that YouTube-VIS 2021 images are being used.
   Do I need to download the YouTube-VIS 2021 dataset or any other image datasets?
   Because in `zhouyik/MMVMData`, there are only JSON files.


2. Is the category needed to format the user question?
   I don't think the category label will be used, so it shouldn't be a problem.


3. There are disappear cases in my annotations
4. Some cases maybe too easy to train your model
Thank you, understood.

**Comment by zhouyiks** [comment]
Here is the code about how I synthesize the QUESTION:
https://github.com/zhouyiks/CoLVA/blob/e0d99c10b80a34b7763c5896b4a7aadd0dc7b600/projects/colva/dataset/map_fns/match_map_fn.py#L153

Yes, SFT data has a different format from the MMVMBench. Basically, it is a difference between MCQ and VQA.

Yes, You need to download these video data. Here is my instruction:
https://github.com/zhouyiks/CoLVA/issues/3
